shm_wrapper/shm_wrapper.c

```diff
@@ -837,12 +859,17 @@ int log_data_write(char* key, param_type_t type, param_val_t value) {
         }
     }
 
-    // return if we didn't find the log data corresponding to this key
+    // return if we ran out of keys for log data
     if (idx == UCHAR_MAX) {
         log_printf(ERROR, "Maximum number of %d log data keys reached. can't add key %s", UCHAR_MAX, key);
         return -1;
     }
 
+    if (strlen(key) >= LOG_KEY_LENGTH) {
+        log_printf(ERROR, "Key name %s for log data is longer than 64 characters", key);
```

Comment:
A string of length `LEN` has `LEN + 1` characters (null terminator), so is `LOG_KEY_LENGTH` the upper bound for the key string? Also we can use the `#define` in the log as well.

Reply:
its the max length including the null byte.